commands/Cancel.java

```diff
@@ -1,41 +1,42 @@
 package org.usfirst.frc4904.standard.commands;
 
+import java.util.Set;
+
 import org.usfirst.frc4904.standard.LogKitten;
-import edu.wpi.first.wpilibj2.command.CommandBase;
 import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Subsystem;
 
-public class Cancel extends CommandBase implements Command
-{
+public class Cancel implements Command {
 	protected final Command command;
 
-	public Cancel(Command command) 
-	{
+	public Cancel(Command command) {
 		this.command = command;
 	}
 
 	@Override
-	public void initialize() 
-	{
+	public void initialize() {
 		LogKitten.v("Initializing " + getName());
 		command.cancel();
 	}
 
 	@Override
-	public void execute() {}
+	public void execute() {
+	}
 
 	@Override
-	public boolean isFinished() 
-	{
+	public boolean isFinished() {
 		return true;
 	}
 
 	@Override
-	protected void end() 
-	{
+	public void end(boolean interrupted) {
+		if(interrupted) {
+			LogKitten.v("Cancel was interrupted");
+		}
 	}
 
 	@Override
-	protected void interrupted() 
-	{
+	public Set<Subsystem> getRequirements() {
+		return null;
```

Comment:
Shouldn't we be returning an empty set rather than null when there are no requirements?

Reply:
I agree—this could cause problems for future higher-level functions when null doesn't have the required set methods.